validations.go

```diff
@@ -52,6 +69,21 @@ type validateSaveable interface {
 }
 
 func (m *Model) validateSave(c *Connection) (*validate.Errors, error) {
+	v := reflect.Indirect(reflect.ValueOf(m.Value))
```

Comment:
I feel like a lot of this logic is repeated. would make sense to break it out into it's own method?